utils/utils.go

```diff
@@ -123,7 +127,18 @@ func ReadPublicURL(url string) ([]byte, error) {
 	}(resp.Body)
 
 	// allow images of up to 1 MiB
-	return io.ReadAll(http.MaxBytesReader(nil, resp.Body, 1*1024*1024))
+	response, err := io.ReadAll(http.MaxBytesReader(nil, resp.Body, httpResponseLimit))
+	if err != nil {
+		// We are doing this because errors.Is(err) check doesn't work for this
+		// since MaxBytesError has pointer receiver. Not sure what is the correct
+		// way to do this.
+		maxByteErr := http.MaxBytesError{}
+		if err.Error() == maxByteErr.Error() {
+			return nil, ErrResponseTooLarge
```

Comment:
(chatGPT suggestions...) can't you do something like:
```
if errors.Is(err, &MaxBytesError{})
```

Reply:
nah. tried. it doesn't work. since it compares the struct which is different